notes/zrq/20210212-01-speed-tests.txt

```diff
@@ -0,0 +1,881 @@
+        Took 18 min 35 sec. Last updated by gaiauser at February 12 2021, 10:08:23 PM.
+
+        Hertzsprung-Russell
+        Took 54 min 22 sec. Last updated by gaiauser at February 12 2021, 11:02:46 PM.
+
+        histogram
+        Took 14 min 58 sec. Last updated by gaiauser at February 12 2021, 11:17:44 PM.
+
+        Good sources plot
+        Took 27 min 12 sec. Last updated by gaiauser at February 12 2021, 11:44:56 PM.
+
+        Bad sources plot
+        Took 27 min 13 sec. Last updated by gaiauser at February 13 2021, 12:12:09 AM.
+
+        No. of good sources:  22254
+        No. of bad sources:   26170
+        Took 27 min 42 sec. Last updated by gaiauser at February 13 2021, 12:39:51 AM.
+
+        histogram
+        Took 19 min 10 sec. Last updated by gaiauser at February 13 2021, 12:59:01 AM.
+
+        Nulls
+        Took 15 min 48 sec. Last updated by gaiauser at February 13 2021, 1:14:49 AM.
+
+        ----
+
+dev deployment #3
+
+    test #3.5
+
+        repeat of the same
+        100% data, 500 trees, no cache
+
+        clear cells and run all
+
+        ml intro
+        Took 0 sec. Last updated by gaiauser at February 13 2021, 3:19:07 AM.
+
+        temp view
+        Took 50 sec. Last updated by gaiauser at February 13 2021, 3:19:57 AM.
+
+        main select statement
+        1724028
+        Took 38 sec. Last updated by gaiauser at February 13 2021, 3:20:35 AM.
+
+        Hertzsprung-Russell
+        Took 4 min 1 sec. Last updated by gaiauser at February 13 2021, 3:24:36 AM.
+
+        good/bad select - 100% data
+        Good training data size: 244740 rows
+        Bad  training data size: 244740 rows
+        Took 7 min 4 sec. Last updated by gaiauser at February 13 2021, 3:31:40 AM.
+
+        RandomForestClassifier - 100% data 500 trees
+        Took 1 hrs 19 min 39 sec. Last updated by gaiauser at February 13 2021, 4:51:20 AM.
+
+        Misclassifications for the test set: 0.35 %
+        Took 20 min 13 sec. Last updated by gaiauser at February 13 2021, 5:11:34 AM.
+
+        Hertzsprung-Russell
+        Took 55 min 7 sec. Last updated by gaiauser at February 13 2021, 6:06:42 AM.
+
+        histogram
+        Took 14 min 12 sec. Last updated by gaiauser at February 13 2021, 6:20:54 AM.
+
+        Good sources plot
+        Took 27 min 15 sec. Last updated by gaiauser at February 13 2021, 6:48:09 AM.
+
+        Bad sources plot
+        Took 27 min 56 sec. Last updated by gaiauser at February 13 2021, 7:16:06 AM.
+
+        good/bad count
+        No. of good sources:  22254
+        No. of bad sources:   26170
+        Took 27 min 16 sec. Last updated by gaiauser at February 13 2021, 7:43:22 AM.
+
+        histogram
+        Took 19 min 43 sec. Last updated by gaiauser at February 13 2021, 8:03:05 AM.
+
+        Nulls
+        Took 15 min 54 sec. Last updated by gaiauser at February 13 2021, 8:18:59 AM.
+
+    TODO
+
+        retry with caching enabled
+        retry with 1000 and 5000 trees
+
+
```

Comment:
Am I correct in thinking we are getting inconsistent & inconclusive results? Looks like the speed of the tests can vary quite a bit, and it looks like they are being affected by something outside of our control?

Reply:
Yes, absolutley.
Turns out the Ceph file store was having to cope with multiple disc failures at the time.
On a good day this notebook is limited by the available read data rate from the Ceph file store, this was a bad day and the number were all over the place.
Following these tests I contacted John Garbutt and he ended up removing a couple of discs and re-building the RAID array.